### PR TITLE
Use calculated values from calc MS

### PR DESCRIPTION
**Pull request by st3v3nhunt**

### CODE REVIEW

File: app/routes/ready-reckoner-v2/calculating.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+function hydrateStandards (baseStandards, selectedStandards, landValues, actionValues) {
+  // Could filter landValues to only those with a value greater than 0 but by
+  // not doing this, they can run through the calculation and everything will
+  // work, it'll be easier to reason about and if they were to be required
+  // within the calculation at some point they are already there
+  const actionValueList = Object.keys(actionValues)
+  return Object.entries(landValues).reduce((acc, cur) => {
+    const [k, v] = cur
+    baseStandards.filter(s => s.landType === k).forEach(s => {
+      s.inputValue = Number(v)
```
Comment (paulsimonandrews) [inline app/routes/ready-reckoner-v2/calculating.js]: Shouldn't need to do this conversion as the Joi validation ensures landValues are always numbers. No need to remove the conversion though as it doesn't hurt.

File: app/routes/ready-reckoner-v2/calculating.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+const { v4: uuid } = require('uuid')
+
+const baseStandards = require('../../services/standards-v2')
+const session = require('./session-handler')
+const { updateAgreement } = require('../../messaging/senders')
+
+// NOTE: map the land types to the standards. doesn't matter which way round as
+//       the relationship is assumed to be 1-to-1
```
Comment (paulsimonandrews) [inline app/routes/ready-reckoner-v2/calculating.js]: Just a note that I see the relationship as many-to-one. 

You can have two standards (eg arable and arable soils) that use the same land value (amount of arable land).

I've not seen any circumstance where a standard uses more than one land value.

File: app/routes/ready-reckoner-v2/sfi-summary.js [hunk 3]
```diff
@@ -96,53 +99,55 @@ function doPaymentCalculations (landValues, actionValues, bpsPayment, selectedSt
...
-      pageDetails.backPath = '/' + request.info.referrer.split('/').slice(-1)[0]
+      pageDetails.backPath = 'javascript:history.go(-1)'
 
-      // Do payment calculation
+      const correlationId = session.getValue(request, session.keys.correlationId)
+      const url = `${agreementServiceBaseUrl}/value?correlationId=${correlationId}`
+      const { payload } = await Wreck.get(url, { json: true })
+      log('msg response', payload)
+
+      // TODO: potentially replace with the data from the msg
```
Comment (paulsimonandrews) [inline app/routes/ready-reckoner-v2/sfi-summary.js:144]: Yeah. I always hate trying to decide where to get data from when there are two sources like this!

File: app/routes/ready-reckoner-v2/sfi-summary.js [hunk 4]
```diff
@@ -159,27 +164,27 @@ module.exports = [
           categoryAmounts[id][feature] = landValues[feature]
           landFeatures[feature].standards.forEach(standard => {
             if (selectedStandards.includes(standard)) {
+              console.log(paymentAmounts.standards[standard])
```
Comment (paulsimonandrews) [inline app/routes/ready-reckoner-v2/sfi-summary.js]: Remove console log

File: app/services/standards-v2.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  }
+}, {
+  id: 'hedgerows',
+  calculations: [{
+    condition: 'true',
+    expression: 'inputValue * paymentRate'
+  }],
+  landType: 'hedgerows',
+  optionalActions: [],
+  paymentRate: 16,
```
Comment (paulsimonandrews) [inline app/services/standards-v2.js]: Current iteration of prototype has a payment rate of £200 per hectare

File: app/services/standards-v2.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  title: 'Hedgerows',
+  units: {
```
Comment (paulsimonandrews) [inline app/services/standards-v2.js:136]: Should be hectares to match version of protoype this is based on

File: app/services/standards-v2.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+  id: 'unimproved-grassland',
+  calculations: [{
+    condition: 'true',
+    expression: 'inputValue * paymentRate'
+  }],
+  landType: 'unimproved-grassland',
+  optionalActions: [],
+  paymentRate: 22,
+  title: 'Semi-improved and unimproved grassland',
+  units: {
```
Comment (paulsimonandrews) [inline app/services/standards-v2.js:122]: Should be hectares

File: app/routes/ready-reckoner-v2/sfi-summary.js [hunk 8]
```diff
@@ -159,27 +164,27 @@ module.exports = [
...
         })
 
         categoryAmounts[id].payment += categoryAmounts[id].paymentOptional
       })
 
-      return h.view(pageDetails.template, pageContent(categoryAmounts, actionValues, paymentAmounts))
-    }
-  },
-  {
-    method: 'POST',
```
Comment (paulsimonandrews) [inline app/routes/ready-reckoner-v2/sfi-summary.js:179]: Removing the `POST` route breaks hitting the continue button on this page, so I'd put it back in.

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Looking good.

Some suggestions and a couple of things to fix:
1. Add the POST route back for the SFI summary page
2. The Hedgerow rate and unit is wrong so the calculation comes back differently

File: app/routes/ready-reckoner-v2/sfi-summary.js (hunk 8 above)
Comment (st3v3nhunt) [reply app/routes/ready-reckoner-v2/sfi-summary.js:179]: Ha! I thought it weird that it existed. You can tell I never proceed 🛑

File: app/routes/ready-reckoner-v2/calculating.js (hunk 2 above)
Comment (st3v3nhunt) [reply app/routes/ready-reckoner-v2/calculating.js]: Yeah, good point. The comment is incorrect, good reason not to have it!
The code handles the relationship of 1-to-many (land to standard).

File: app/routes/ready-reckoner-v2/sfi-summary.js (hunk 3 above)
Comment (st3v3nhunt) [reply app/routes/ready-reckoner-v2/sfi-summary.js:144]: I think we should be using the data from the msg but didn't do it as I didn't want to do the mapping from the msg plus there might be a case to use the data we've got whilst waiting for the msg to be returned with the changes to the interface.

**Review by paulsimonandrews — DISMISSED** [review]
LGTM

**Review by paulsimonandrews — APPROVED** [review]
LGTM

**Pull request merged**